src/declare.rs

```diff
@@ -80,9 +80,13 @@ impl<'gcc, 'tcx> CodegenCx<'gcc, 'tcx> {
     }
 
     pub fn declare_fn(&self, name: &str, fn_abi: &FnAbi<'tcx, Ty<'tcx>>) -> Function<'gcc> {
-        let (return_type, params, variadic, on_stack_param_indices) = fn_abi.gcc_type(self);
+        let (return_type, params, variadic, on_stack_param_indices, fn_attrs) = fn_abi.gcc_type(self);
         let func = declare_raw_fn(self, name, () /*fn_abi.llvm_cconv()*/, return_type, &params, variadic);
         self.on_stack_function_params.borrow_mut().insert(func, on_stack_param_indices);
+        // We need to handle `nonnull` here where we still have access to function args.
```

Comment:
Literally the function arguments?

Reply:
Sure, but we also had access to the arguments earlier, so I'm not sure what this comment brings. Plus, I'm not sure why we would need the arguments since the attributes are already computed.

Reply:
Comment might need improvement indeed. I was trying to explanation that we now have the "computed" attributes of the function and that `fn_attributes` is only "available" at this point and if we don't add them to `func`, it will be lost afterward.